plutus-metatheory/src/Algorithmic/CEKC.lagda.md

```diff
@@ -95,6 +97,10 @@ BUILTIN verifySignature σ ((_ ,, _ ,, V-con (bytestring k) ,, _) ,, (_ ,, _ ,,
 BUILTIN equalsByteString σ ((_ ,, _ ,, V-con (bytestring b) ,, _) ,, (_ ,, _ ,, V-con (bytestring b') ,, _) ,, tt) = _ ,, con (bool (equals b b')) ,, []
 BUILTIN ifThenElse σ ((_ ,, _ ,, V-con (bool true) ,, _) ,, (_ ,, t ,, _ ,, ρ) ,, _ ,, tt) = _ ,, t ,, ρ
 BUILTIN ifThenElse σ ((_ ,, _ ,, V-con (bool false) ,, _) ,, _ ,, (_ ,, u ,, _ ,, ρ) ,, tt) = _ ,, u ,, ρ
+BUILTIN charToString _ ((_ ,, _ ,, V-con (char c) ,, _) ,, tt) = _ ,, con (string (primStringFromList L.[ c ])) ,, []
```

Comment:
@jmchapman have you considered deriving all of that automatically given some data type for representing types of builtins? It's just that there are sooo many various ways to assign a semantics to a builtin in `plutus-metatheory` and all of that code is basically boilerplate.

Reply:
https://jira.iohk.io/browse/SCP-1432